backend/infrahub/core/validators/node/inherit_from.py

```diff
@@ -37,12 +38,32 @@ async def check(self, request: SchemaConstraintValidatorRequest) -> list[Grouped
             name=request.node_schema.kind, branch=request.branch, duplicate=False
         )
 
-        if not isinstance(request.node_schema, NodeSchema):
+        if not isinstance(request.node_schema, NodeSchema) or request.schema_branch is None:
             return grouped_data_paths_list
 
         _, removed, _ = compare_lists(list1=current_schema.inherit_from, list2=request.node_schema.inherit_from)
-
-        if removed:
+        current_inherit_from_ids = {
+            g.id: g.kind
+            for g in [
+                self.db.schema.get(name=n, branch=request.branch, duplicate=False) for n in current_schema.inherit_from
+            ]
+        }
+
+        # Gather IDs for each inherited node in use for candidate schema
+        request_inherited: list[MainSchemaTypes] = []
+        for n in request.node_schema.inherit_from:
+            try:
+                request_inherited.append(request.schema_branch.get(name=n, duplicate=False))
+            except SchemaNotFoundError:
+                request_inherited.append(self.db.schema.get(name=n, branch=request.branch, duplicate=False))
+        request_inherit_from_ids = {g.id: g.kind for g in request_inherited}
```

Comment:
For readability `request_inherited.append` could be performed after the `try except` block